### PR TITLE
Bug fixes in the Add subcommand

### PR DESCRIPTION
**Pull request by Rithesh17**
- Any errors in Add will not add the entry to index
- Infers source types from URIs. Throws an error if inconsistent source types in URIs
- Search now works for Webpages
- Fixed the unit testcases - all passed

### CODE REVIEW

File: src/subcommand.go [hunk 1]
```diff
@@ -289,3 +297,17 @@ func CommandReadRun() {
 func loadIndex(indexDir string) {
 	fmt.Println("loadIndex in commands.go is unimplemented")
 }
+
+func identifySource(uris []string) (string, error) {
```
Comment (fredxfred) [inline src/subcommand.go]: let's name this like inferSourceType

File: src/subcommands/utils.go [hunk 2]
```diff
@@ -168,6 +168,23 @@ func fetchFromCopy(indexPath string, name string) ([]byte, error) {
 	return []byte(ile.Content), nil
 }
 
+func createSearchDictionary(ile *pb.IndexListEntry) error {
```
Comment (fredxfred) [inline src/subcommands/utils.go]: Why are we moving this into utils again? I think it should stay in lexical_search.go or go somewhere else dedicated to search

Comment (Rithesh17) [reply src/subcommands/utils.go]: createSearchDictionary function is called in the Add subcommand. And the word occurrences are written into the index list. So it's not exclusive for lexical search.

File: src/subcommand.go (hunk 1 above)
Comment (Rithesh17) [reply src/subcommand.go]: Done

File: src/subcommands/utils.go (hunk 2 above)
Comment (fredxfred) [reply src/subcommands/utils.go]: Let's add it to a new file like src/search/some_file.go

Comment (fredxfred) [reply src/subcommands/utils.go]: In general we should always avoid adding stuff to utils and what we do add should be purely simple one-off stuff, not core functionaltiy

Comment (Rithesh17) [reply src/subcommands/utils.go]: Adding a new file makes more sense. We might have to add more functions later as we add semantic search.

**Pull request merged**